### PR TITLE
Zth iteration4

### PR DESCRIPTION
**Pull request by zachholcomb**
Add helper method to get season tackles in season stat team
Add tackles to team_data hash in season stat team
Add most tackles to season stat team and stat tracker
Add fewest tackles to season stat team and stat tracker
Passing all tests and spec harness
Spec harness is up to 20 seconds so needs refactoring to speed up time

### CODE REVIEW

File: lib/season_stat_team.rb [hunk 1]
```diff
@@ -28,6 +28,16 @@ def get_team_name(team_id)
     team_name_by_id.team_name
   end
 
+  def get_tackles_by_team_season(team_id, season)#refactor to a reduce
+    team_tackles = 0
+    get_season_game_teams(season).each do |game_team|
+      if game_team.team_id.to_s == team_id
+        team_tackles += game_team.tackles
+      end
```
Comment (zachholcomb) [inline lib/season_stat_team.rb:36]: another helper method that is similar to all others, this grabs the tackles and throws it in an accumulator

File: lib/season_stat_team.rb [hunk 2]
```diff
@@ -58,7 +68,8 @@ def create_team_data_by_season(season)
     get_team_ids_by_season(season).reduce({}) do |acc, team_id|
       acc[team_id.to_s] = {
         team_name: get_team_name(team_id),
-        goal_ratio: team_shots_to_goal_ratio_by_season(team_id.to_s, season)
+        goal_ratio: team_shots_to_goal_ratio_by_season(team_id.to_s, season),
+        tackles: get_tackles_by_team_season(team_id.to_s, season)
```
Comment (zachholcomb) [inline lib/season_stat_team.rb:72]: added the tackles to my team_data hash

File: lib/season_stat_team.rb [hunk 3]
```diff
@@ -77,4 +88,18 @@ def least_accurate_team(season)
...
+    end
+    team_tackles[1][:team_name]
+  end
+
+  def fewest_tackles(season)
+    team_tackles = create_team_data_by_season(season).min_by do |team, team_data|
+      team_data[:tackles]
+    end
+    team_tackles[1][:team_name]
+  end
```
Comment (zachholcomb) [inline lib/season_stat_team.rb:104]: used max and min by respectively to grab most and fewest tackles

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -118,4 +118,14 @@ def least_accurate_team(season_param)
...
+  def most_tackles(season_param)
+    season = SeasonStatTeam.new(@game_teams_file, @teams_file)
+    season.most_tackles(season_param)
+  end
+
+  def fewest_tackles(season_param)
+    season = SeasonStatTeam.new(@game_teams_file, @teams_file)
+    season.fewest_tackles(season_param)
+  end
+
```
Comment (zachholcomb) [inline lib/stat_tracker.rb:130]: added them to stat_tracker to check rspec, passing tests

File: lib/season_stat_team.rb (hunk 1 above)
Comment (StephanieFriend) [reply lib/season_stat_team.rb:36]: I already like the way this looks, but make sure we get rid of comments before we push the final product.

File: lib/season_stat_team.rb (hunk 3 above)
Comment (StephanieFriend) [reply lib/season_stat_team.rb:104]: I like the way this is set up. Very clean and neat

**Review by StephanieFriend — APPROVED** [review]
Overall this looks great! I see even more similar helper methods in this PR to my own code. We should be able to get that 20 seconds down. :)

**Pull request merged**